chainstate/src/detail/chainstateref/mod.rs

```diff
@@ -423,55 +426,49 @@ impl<'a, S: BlockchainStorageRead, V: TransactionVerificationStrategy> Chainstat
         Ok(())
     }
 
-    fn check_block_reward_timelock(
-        &self,
-        block: &Block,
-        timelock: &OutputTimeLock,
-    ) -> Result<(), CheckBlockError> {
-        match timelock {
-            OutputTimeLock::ForBlockCount(c) => {
-                let cs: i64 = (*c)
-                    .try_into()
-                    .map_err(|_| {
-                        CheckBlockError::InvalidBlockRewardMaturityDistanceValue(block.get_id(), *c)
-                    })
-                    .log_err()?;
-                let given = BlockDistance::new(cs);
-                let required = block.consensus_data().reward_maturity_distance(self.chain_config);
-                ensure!(
-                    given >= required,
-                    CheckBlockError::InvalidBlockRewardMaturityDistance(
-                        block.get_id(),
-                        given,
-                        required
-                    )
-                );
-                Ok(())
-            }
-            OutputTimeLock::UntilHeight(_)
-            | OutputTimeLock::UntilTime(_)
-            | OutputTimeLock::ForSeconds(_) => Err(
-                CheckBlockError::InvalidBlockRewardMaturityTimelockType(block.get_id()),
-            ),
-        }
-    }
-
     fn check_block_reward_maturity_settings(&self, block: &Block) -> Result<(), CheckBlockError> {
-        block.block_reward().outputs().iter().try_for_each(|output| {
-            match output {
-                TxOutput::LockThenTransfer(_, _, tl) => self.check_block_reward_timelock(block, tl),
-                TxOutput::ProduceBlockFromStake(_, _) => {
-                    // The output can be reused in block reward right away
-                    Ok(())
+        block
+            .block_reward()
+            .outputs()
+            .iter()
+            .enumerate()
+            .try_for_each(|(index, output)| {
+                match block.consensus_data() {
+                    ConsensusData::None | ConsensusData::PoW(_) => match output {
```

Comment:
Do we want to put block reward limitations on ConsensusData::None?

Reply:
I'd say yes, but the maturity distance is 0